float32 cross_track_error 
float32 heading_error